File: data_structure/Kadane's Algorithm.java
class Solution{

    // arr: input array
    // n: size of array
    //Function to find the sum of contiguous subarray with maximum sum.
    long maxSubarraySum(int arr[], int n){
        
        // Your code here
        long sum=arr[0];
        long currsum=arr[0];
        for(int i=1;i<n;i++)
        {
            currsum=Math.max(currsum+arr[i],arr[i]);
            sum=Math.max(sum,currsum);
        }
        return sum;
        
    }
    
}

File: LeetCodeQuestions/Kadane'sAlgorithm.cpp
#include <bits/stdc++.h>
using namespace std;

/*Given an integer array arr, find the 
subarray with the largest sum, and return its sum.*/

long long maxSubarraySum(int arr[], int n)
{
    long long int ans=0, sum=0;
    for(int i=0;i<n;i++)
    {
        sum+=arr[i];
        if(sum<0)
            sum=0;
        ans = max(ans, sum);
    }
    return ans;
}

int main()
{
    cout<<"Enter the number of elements:"<<endl;
    int n;
    cin>>n;
    cout<<"Enter the elements:"<<endl;
    int arr[n];
    for(int i=0;i<n;i++)
        cin>>arr[i];
    long long answer = maxSubarraySum(arr, n);
    cout<<"The maximum subarray sum is:"<<endl;
    cout<<answer<<endl;
}